packages/core/contracts-ovm/insured-bridge/implementation/BridgePoolFactory.sol

```diff
@@ -0,0 +1,187 @@
+     * contract.
+     * @dev Can only be called by the current owner.
+     * @param _identifier New identifier to set.
+     */
+    function setIdentifier(bytes32 _identifier) public onlyOwner {
+        _setIdentifier(_identifier);
+    }
+
+    /**
+     * @notice Sets challenge period for relayed deposits. BridgePools will read this value from this
+     * contract.
+     * @dev Can only be called by the current owner.
+     * @param _liveness New OptimisticOracle liveness period to set for relay price requests.
+     */
+    function setOptimisticOracleLiveness(uint256 _liveness) public onlyOwner {
+        _setOptimisticOracleLiveness(_liveness);
+    }
+
+    /**
+     * @notice Sets challenge pereiod for relayed deposits. BridgePools will read this value from this
+     * contract.
+     * @dev Can only be called by the current owner.
+     * @param _proposerBondPct New OptimisticOracle proposer bond % to set for relay price requests. 1e18 = 100%.
+     */
+    function setProposerBondPct(uint256 _proposerBondPct) public onlyOwner {
+        _setProposerBondPct(_proposerBondPct);
+    }
+
+    /**
+     * @notice Privileged account can set L2 deposit contract that originates deposit orders to be fulfilled by this
+     * contract.
+     * @dev Only callable by the current owner.
+     * @param _depositContract Address of L2 deposit contract.
+     */
+    function setDepositContract(address _depositContract) public onlyOwner {
+        _validateDepositContract(_depositContract);
+        depositContract = _depositContract;
+        emit SetDepositContract(depositContract);
+    }
+
+    /**
+     * @notice Privileged account can associate a whitelisted token with its linked token address on L2 and its
+     * BridgePool address on this network. The linked L2 token can thereafter be deposited into the Deposit contract
+     * on L2 and relayed via the BridgePool contract.
+     * @dev Only callable by Owner of this contract. Also initiates a cross-chain call to the L2 Deposit contract to
+     * whitelist the token mapping.
+     * @param _l1Token Address of L1 token that can be used to relay L2 token deposits.
+     * @param _l2Token Address of L2 token whose deposits are fulfilled by `_l1Token`.
+     * @param _bridgePool Address of pool contract that stores passive liquidity with which to fulfill deposits.
+     * @param _l2Gas Gas limit to set for relayed message on L2
+     */
+    function whitelistToken(
+        address _l1Token,
+        address _l2Token,
+        address _bridgePool,
+        uint32 _l2Gas
+    ) public onlyOwner depositContractSet {
+        require(_getCollateralWhitelist().isOnWhitelist(address(_l1Token)), "Payment token not whitelisted");
+
+        L1TokenRelationships storage whitelistedToken = whitelistedTokens[_l1Token];
+        whitelistedToken.l2Token = _l2Token;
+        sendCrossDomainMessage(
+            depositContract,
+            _l2Gas,
```

Comment:
What happens if the gas is too low and the OVM transaction fails?

Reply:
Good point. I'll add a todo. we should prepare for that

Reply:
Currently, the user could simply call the function again.